### PR TITLE
JDK-8301508: Replace NULL with nullptr in os_cpu/linux_s390

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory os_cpu/linux_s390. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301508](https://bugs.openjdk.org/browse/JDK-8301508): Replace NULL with nullptr in os_cpu/linux_s390

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12337/head:pull/12337` \
`$ git checkout pull/12337`

Update a local copy of the PR: \
`$ git checkout pull/12337` \
`$ git pull https://git.openjdk.org/jdk pull/12337/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12337`

View PR using the GUI difftool: \
`$ git pr show -t 12337`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12337.diff">https://git.openjdk.org/jdk/pull/12337.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os_cpu/linux_s390/javaThread_linux_s390.cpp [hunk 1]
```diff
@@ -60,17 +60,17 @@ bool JavaThread::pd_get_top_frame_for_profiling(frame* fr_addr, void* ucontext,
...
       return false;
     }
 
     frame ret_frame((intptr_t*)uc->uc_mcontext.gregs[15/*Z_SP*/], pc);
 
-    if (ret_frame.fp() == NULL) {
+    if (ret_frame.fp() == nullptr) {
       // The found frame does not have a valid frame pointer.
       // Bail out because this will create big trouble later on, either
-      //  - when using istate, calculated as (NULL - z_ijava_state_size (= 0x70 (dbg) or 0x68 (rel)) or
```
Comment (jdksjolen) [inline src/hotspot/os_cpu/linux_s390/javaThread_linux_s390.cpp:73]: nullptr

File: src/hotspot/os_cpu/linux_s390/javaThread_linux_s390.cpp [hunk 2]
```diff
@@ -92,9 +92,9 @@ bool JavaThread::pd_get_top_frame_for_profiling(frame* fr_addr, void* ucontext,
       // contents of r6:  0xffffffffffffff90
       //
       // Here is the sequence of what happens:
-      //  - ret_frame is constructed with _fp == NULL (for whatever reason)
+      //  - ret_frame is constructed with _fp == null (for whatever reason)
       //  - ijava_state_unchecked() calculates it's result as
-      //      istate = fp() - z_ijava_state_size() = NULL - 0x68 DEBUG_ONLY(-8)
+      //      istate = fp() - z_ijava_state_size() = null - 0x68 DEBUG_ONLY(-8)
```
Comment (jdksjolen) [inline src/hotspot/os_cpu/linux_s390/javaThread_linux_s390.cpp:97]: nullptr

File: src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp [hunk 3]
```diff
@@ -190,9 +190,9 @@ frame os::current_frame() {
     return frame();
   } else {
     frame senderFrame = os::get_sender_for_C_frame(&topframe);
-    assert(senderFrame.pc() != NULL, "Sender pc should not be NULL");
+    assert(senderFrame.pc() != nullptr, "Sender pc should not be null");
     // Return sender of sender of current topframe which hopefully
-    // both have pc != NULL.
+    // both have pc != null.
```
Comment (jdksjolen) [inline src/hotspot/os_cpu/linux_s390/os_linux_s390.cpp:195]: nullptr

**Review by jdksjolen — COMMENTED** [review]
Some fixes, otherwise trivial.

**Review by kimbarrett — APPROVED** [review]
Looks good, and trivial.

**Comment by jdksjolen** [comment]
Thanks!

/integrate

**Pull request closed without merging**